src/arrayMethodSort.js

```diff
@@ -5,8 +5,63 @@
  */
 function applyCustomSort() {
   [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+    let currentValue;
+    let previousValue;
+    let isSorted;
+
+    if (compareFunction !== undefined) {
+      while (!isSorted) {
+        isSorted = true;
+
+        for (let i = 1; i < this.length; i++) {
+          previousValue = this[i - 1];
+          currentValue = this[i];
```

Comment:
```suggestion
          const previousValue = this[i - 1];
          const currentValue = this[i];
```
As you don't use these values outside the loop you can create them right here